bio-qiime-pipeline/src/primer_trimming.py

```diff
@@ -0,0 +1,98 @@
+from pathlib import Path
+from zipfile import ZipFile, ZIP_DEFLATED
+
+import logging
+
+from coretex import Experiment, CustomDataset, CustomSample, folder_manager
+from coretex.bioinformatics import cutadaptTrim
+
+from .utils import loadPairedEnd
+
+
+def forwardMetadata(sample: CustomSample, outputDataset: CustomDataset) -> None:
+    sample.unzip()
+    metadataZip = folder_manager.temp / "metadata.zip"
+    with ZipFile(metadataZip, 'w', ZIP_DEFLATED) as archive:
+        for path in sample.path.iterdir():
+            archive.write(path, path.name)
+
+    if CustomSample.createCustomSample(sample.name, outputDataset.id, metadataZip) is None:
+        raise RuntimeError(">> [Microbiome analysis] Failed to forward metadata to the output dataset")
+
+
+def loadSingleEnd(sample: CustomSample) -> Path:
+    sample.unzip()
+    for filePath in sample.path.iterdir():
+        if filePath.suffix != ".fastq":
+            continue
+
+        return filePath
+
+    raise ValueError(f">> [Microbiome analysis] Sample \"{sample.name}\" does not contain fastq files")
+
+
+def uploadSingle(file: Path, dataset: CustomDataset):
+    zipPath = folder_manager.temp / f"{file.name}.zip"
+    with ZipFile(zipPath, 'w', ZIP_DEFLATED) as archive:
+        archive.write(file, file.name)
+
+    if CustomSample.createCustomSample(file.name, dataset.id, zipPath) is None:
+        raise RuntimeError(">> [Microbiome analysis] Failed to upload trimmed reads")
+
+
+def uploadPaired(forwardFile: Path, reverseFile: Path, sampleName: str, dataset: CustomDataset):
```

Comment:
`uploadSingle` and `uploadPaired` can be one function where the `reverseFile` is an optional property and is placed into zip only if it exists